File: app-budget-overrun-tracker/src/main/java/com/github/arhor/aws/microservices/playground/overruns/data/model/BudgetOverrunDetails.java
package com.github.arhor.aws.microservices.playground.overruns.data.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BudgetOverrunDetails {
    private Long userId;
    private BigDecimal amount;
}

File: app-expenses/src/main/kotlin/com/github/arhor/aws/microservices/playground/expenses/data/model/projection/BudgetOverrunDetails.kt
package com.github.arhor.aws.microservices.playground.expenses.data.model.projection

import org.springframework.data.annotation.PersistenceCreator
import java.math.BigDecimal

data class BudgetOverrunDetails @PersistenceCreator constructor(
    val userId: Long,
    val amount: BigDecimal,
) {
    internal constructor(userId: Long, amount: String) : this(userId, BigDecimal(amount))
}
